### PR TITLE
test: check status code in specs for PlotlyAPI

### PR DESCRIPTION
**Pull request by n-riesco**
@shannonlal I'd like to use this PR to show you two of the helper functions we use in the specs (`assertResponseStatus` and `getResponseJson`). Would  you like to review this PR?

When we review a PR, we have the convention of using 💃 to indicate the points raised by the review have been addressed and the PR is ready to merge.

### CODE REVIEW

**Comment by shannonlal** [comment]
@n-riesco Looking at this now.

File: test/backend/PlotlyAPI.spec.js [hunk 1]
```diff
@@ -24,31 +31,29 @@ describe('Grid API Functions', function () {
         // it works off of the assumption that a grid exists
 
         let fid;
-        return createGrid('Test updateGrid').then(res => res.json().then(json => {
-            // Update the grid's data
+        return createGrid('Test updateGrid')
+        .then(assertResponseStatus(201))
```
Comment (shannonlal) [inline test/backend/PlotlyAPI.spec.js]: Should 201 be defined as a comment?

File: test/backend/PlotlyAPI.spec.js [hunk 2]
```diff
@@ -24,31 +31,29 @@ describe('Grid API Functions', function () {
...
+                    ['x', 10, 40, 70, 100, 130],
+                    ['y', 20, 50, 80, 110, 140],
+                    ['z', 30, 60, 90, 120, 150]
+                ],
+                fid,
+                uids,
+                username
+            );
+        })
+        .then(assertResponseStatus(200))
```
Comment (shannonlal) [inline test/backend/PlotlyAPI.spec.js]: This may be a small thing.  But wondering if we should use a constant here

**Comment by shannonlal** [comment]
@n-riesco I added some small comments on this PR.  They are minor but I thought I would get your input first.  You can message me during the day and if you want I can Merge the PR.

**Comment by n-riesco** [comment]
@shannonlal 

> Some of the tests are named lower case and some are named up case. Is there a convention?

No, we don't have a convention, but I usually follow this convention:
```js
describe('tag:', function() {
    describe('subtag:', function() {
        it('myFunction should do this', function() {
            assert(condition, 'Unmet condition');
        });
    });
});
```

> You can message me during the day and if you want I can Merge the PR.

No worries, there's not rush.

I'm also used to having the reviewer merge the PR, but we don't follow that convention in this repository.

The author of the PR merges the PR, once the reviewer gives a :dancer: . Only, if the author of the PR doesn't have merge rights, then the reviewer merges it.

File: test/backend/PlotlyAPI.spec.js (hunk 1 above)
Comment (n-riesco) [reply test/backend/PlotlyAPI.spec.js]: I've added a link to the documentation of Plot.ly's REST API

File: test/backend/PlotlyAPI.spec.js (hunk 2 above)
Comment (n-riesco) [reply test/backend/PlotlyAPI.spec.js]: I wouldn't define constants for this. Typically, I'd define a constant to make the code more readable. In this case, the function name `assertResponseStatus(200)` is enough to tell the reader 200 is a status code.

**Comment by shannonlal** [comment]
💃

**Pull request merged**